### PR TITLE
Extern functionals

### PR DESCRIPTION
**Pull request by gonzalodm**
Now you can use more fuctionales in LIO. You need to compile LIO with libxc and libint libraries.
The new functionales are: b3lyp, lc_wpbe, lc_blyp and cam-b3lyp. It's easy to add new functionales (I wil write the steps in order to add new functionales). This implementation only works with the latest version of libxc. For the moment, these new functionales only works on CPU in Ground and Excited states calculations. In the next PR, I'm going to make it work in TD. 
PD: This implementation in GPU is impossible for the moment, because the version of libxc-GPU is older and does not have the range-correction implemented.

### CODE REVIEW

**Comment by nanolebrero** [comment]
Imposible is nothing.

File: lioamber/excitedsubs/PCG_solve.f90 [hunk 1]
```diff
@@ -175,11 +175,11 @@ subroutine Ap_calculate(Fe,Fx,P,E,Ap,M,Mlr,NCO,Nvirt,Ndim)
 
    ! Obtain total fock
    allocate(Fp(M,M))
-   do ii=1,M
```
Comment (fedepedron) [inline lioamber/excitedsubs/PCG_solve.f90:178]: Remove commented out code.

File: lioamber/extern_functional.f90 [hunk 2]
```diff
@@ -0,0 +1,259 @@
+#include "datatypes/datatypes.fh"
+! This module permites the use of functionals from libxc
```
Comment (fedepedron) [inline lioamber/extern_functional.f90:2]: "enables the usage" instead of "permites the use".

File: lioamber/liomods/garcha_mod.f90 [hunk 3]
```diff
@@ -55,8 +55,7 @@ module garcha_mod
    integer, allocatable, dimension(:) ::  restr_index
    LIODBLE, allocatable, dimension(:) :: restr_k, restr_w, restr_r0
 
-   ! For pbe0 functional
-   logical :: PBE0 = .false.
+   logical :: PBE0 = .false. ! TODO borrar
```
Comment (fedepedron) [inline lioamber/liomods/garcha_mod.f90:58]: Remove?

**Comment by fedepedron** [comment]
Also:
* I get a lot of "unused variable" and "&" missing, and even a conversion warning. Please compile with analytics=3 and fix all warnings related to these changes. 
* A test to verify against it would be great.

**Pull request merged**